src/lite-yt-embed.js

```diff
@@ -52,6 +52,13 @@ class LiteYTEmbed extends HTMLElement {
         // TODO: In the future we could be like amp-youtube and silently swap in the iframe during idle time
         //   We'd want to only do this for in-viewport or near-viewport ones: https://github.com/ampproject/amphtml/pull/5003
         this.addEventListener('click', this.addIframe);
+
+
+        // Chrome & Edge have no problem with the basic YouTube Embed with ?autoplay=1
+        // However Safari and Firefox do not successfully track the user gesture of clicking through the creation/loading of the iframe,
+        // so they don't autoplay automatically. Instead we must load an additional 300KB (ungz) of JS for the YT Player API
+        // TODO: chrome android seems to also need this
+        this.needsYTApiForAutoplay = navigator.vendor.includes('Apple') || navigator.userAgent.includes('Firefox');
```

Comment:
i figure ill have to change this to be some other sniff. don't think there's a good "feature detect" that's equivalent to failing to successfully autoplay. so yeah.. we'll do a UA sniff of some sort.  

but this'll need to also include chrome android as well. and perhaps more.   

some testing required:

#### has problems with autoplay on https://lite-youtube-embed-git-master-paulirish.vercel.app/

ios safari
desktop safari
firefox android
chrome android

#### no problems

desktop chrome

#### ???

firefox desktop
edge desktop and mobile